Mage/src/main/java/mage/abilities/common/SagaAbility.java

```diff
@@ -129,25 +141,71 @@ public static boolean isChapterAbility(TriggeredAbility ability) {
     }
 }
 
+class SagaLoreCountersEffect extends OneShotEffect {
+
+    private final boolean readAhead;
+    private final SagaChapter maxChapter;
+
+    SagaLoreCountersEffect(boolean readAhead, SagaChapter maxChapter) {
+        super(Outcome.Benefit);
+        this.readAhead = readAhead;
+        this.maxChapter = maxChapter;
+    }
+
+    private SagaLoreCountersEffect(final SagaLoreCountersEffect effect) {
+        super(effect);
+        this.readAhead = effect.readAhead;
+        this.maxChapter = effect.maxChapter;
+    }
+
+    @Override
+    public SagaLoreCountersEffect copy() {
+        return new SagaLoreCountersEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent permanent = game.getPermanentEntering(source.getSourceId());
+        if (permanent == null) {
+            return false;
+        }
+        if (!readAhead) {
+            return permanent.addCounters(CounterType.LORE.createInstance(), source, game);
+        }
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+        int counters = player.getAmount(
+                1, maxChapter.getNumber(),
+                "Choose the number of lore counters to enter with", game
+        );
```

Comment:
Is there a way to implement this so that the player is choosing the chapter on which the saga will start on? With the current implementation it looks like the player would have to remember/look up how many counters corresponds to which effect

Reply:
I think this can be done but maybe at a later date since it's not 100% necessary for the implementation